appIconIndicators.js

```diff
@@ -840,6 +870,80 @@ class UnityIndicator extends IndicatorBase {
         this._progressOverlayArea?.queue_repaint();
     }
 
+    _readGradientData(node, elementName, defaultValues) {
+        const output = {
+            colorStart: defaultValues.colorStart,
+            colorEnd: defaultValues.colorEnd,
+            offsetStart: defaultValues.offsetStart ?? 0.0,
+            offsetEnd: defaultValues.offsetEnd ?? 1.0,
+        };
+
+        let hasValue, value1, value2;
+
+        [hasValue, value1] = node.lookup_color(elementName, false);
+        if (hasValue) {
+            output.colorStart = value1;
+            output.colorEnd = null;
+            console.log(value1.red);
+        } else {
+            [hasValue, value1] = node.lookup_color(`${elementName}-color-start`, false);
+            if (hasValue) {
+                [hasValue, value2] = node.lookup_color(`${elementName}-color-end`, false);
+                if (hasValue) {
+                    output.colorStart = value1;
+                    output.colorEnd = value2;
+                }
+            }
```

Comment:
```suggestion
            const [hasStart, colorStart] = node.lookup_color(`${elementName}-color-start`, false);
            const [hasEnd, colorEnd] = node.lookup_color(`${elementName}-color-end`, false);
            if (hasStart && hasEnd)
                output = {...output, colorStart, colorEnd};
```

Reply:
Done, with the exception of the ellipsis...